index.html

```diff
@@ -0,0 +1,90 @@
+<!doctype html>
+<html lang="en">
+<head>
+  <meta charset="utf-8">
+  <title>Random Background Color</title>
+  <script>
+
+      function setValue(v){
+          if (typeof(Storage) !== "undefined") {
+              var listOfValues = getUsedValues();
+              listOfValues.push(v);
+              localStorage.setItem("used", JSON.stringify(listOfValues));
+          }else{
+              used.push(v);
+          }
+      }
+
+
+      function getUsedValues(){
+          if (typeof(Storage) !== "undefined") {
+              var a=  JSON.parse(localStorage.getItem("used")) || [];
+              console.log(a);
+              return a;
+          }else{
+              return []
+          }
+      }
+
+      var used = getUsedValues();
+
+
+
+      var limit = 10;
+      function getRandomColor() {
+          var length = 6;
+          var chars = '0123456789ABCDEF';
+          var hex = '#';
+          while (length--) hex += chars[(Math.random() * 16) | 0];
+          if (getUsedValues().indexOf(hex) == -1) {
+              setValue(hex);
+              return hex;
+          }
+          return getRandomColor();
```

Comment:
Cons
- It might cause stack/inefficient memory issues
- getUsedValues function can cause problems

We could also implement tail call optimization with ES6

Pros
- Less coding 
- It can be efficient with memoization method